lib/qiita/markdown/filters/user_input_sanitizer.rb

```diff
@@ -2,30 +2,90 @@ module Qiita
   module Markdown
     module Filters
       # Sanitizes raw markdown input if :strict context is given.
-      class UserInputSanitizer < HTML::Pipeline::TextFilter
+      class UserInputSanitizer < HTML::Pipeline::Filter
+        class AttributeTransformer
+          FILTERS = {
+            "a" => {
+              "class" => %w[autolink],
+              "rel" => %w[footnote url],
+              "rev" => %w[footnote],
+            },
+            "sup" => {
+              "id" => /^fnref\d+$/,
+            },
+            "li" => {
+              "id" => /^fn\d+$/,
+            },
+          }.freeze
+
+          def self.call(*args)
+            new(*args).transform
+          end
+
+          def initialize(env)
+            @env = env
+          end
+
+          def transform
+            return unless FILTERS.key?(name)
+            FILTERS[name].each_pair do |attr, pattern|
+              transform_attribute(attr, pattern) if node.attributes.key?(attr)
+            end
+          end
+
+          private
+
+          def transform_attribute(attr, pattern)
+            node.attributes[attr].value = node.attributes[attr].value.split.select do |value|
```

Comment:
Omitting argument of `String#split` might be dangerous, since it depends of a special global variable `$;`:

http://ruby-doc.org/core-2.4.1/String.html#method-i-split

> If pattern is nil, the value of $; is used. If $; is nil (which is the default), str is split on whitespace as if ‘ ’ were specified.

Since we're doing secure things here, it would be better to always be defensive 😉